pkg/route/ingress/ingress_test.go

```diff
@@ -888,6 +909,84 @@ func TestController_sync(t *testing.T) {
 				},
 			},
 		},
+		{
```

Comment:
There will be an update to add the annotation but the TLS spec won't be modified. Is this the desired behavior or it's preferable to prevent the annotation sync if the termination type is not reencrypt?

Reply:
Hmm I see your updates. I have to think about that, but the reason I brought up is because of your change in `tlsConfigForIngress()`:
```
	destinationCACertificate := destinationCACertificateForIngress(ingress, secretLister)
	if terminationPolicy == routev1.TLSTerminationReencrypt && destinationCACertificate != nil {
		tlsConfig.DestinationCACertificate = *destinationCACertificate
	}
```

I believe translates to: new routes with any type other than type "reencrypt" will not have their TLS spec updated with `DestinationCACertificate`. That's why I brought up the idea of these tests.

I think I agree with your update. It's basically dropping `DestinationCACertificate` from the route if has been updated to be anything other than re-encrypt. I think it's important to be consistent either way. You won't create `DestinationCACertificate` for brand new routes with type passthrough or edge, so it makes sense you would drop it if it switch from reencrypt to passthrough or edge.

Reply:
But thinking about @Miciah suggestions, he wants to make sure we preserve `DestinationCACertificate` if the user adds it to the route themselves, I start leaning the other way (don't clear `DestinationCACertificate` if a user defines it and changes the termination type to something other than reencrypt). It might be bad UX for the user to put a `DestinationCACertificate` on the route only for it to disappear.

@Miciah could you weigh in? If we switch the termination to something other than , should we clear the route's `DestinationCACertificate`?

Reply:
Good question.  Yeah, if we change the termination type from "reencrypt", then we need to clear `spec.tls.destinationCACertificate` so as not to get a validation error: https://github.com/openshift/openshift-apiserver/blob/db22c038b9184b4ae823bba3f57b669526999a2b/pkg/route/apis/route/validation/validation.go#L253-L260